crypto/dilithium/README.md

```diff
@@ -0,0 +1,71 @@
+
+The AWS-LC team considers the official repository of [Dilithium](https://github.com/pq-crystals/dilithium) the primary source of Dilithium’s implementation and takes the code directly from it. The code is integrated in AWS-LC with only minimal changes that are required to build on the platforms AWS-LC supports (see below for details).
+
+NIST has not published the final PQ standard yet, and is not expected to do so until 2024. Therefore, the specification and implementation of Dilithium is not finalized yet. Potentially, there will be changes to Dilithium in the future. Some changes might even break backwards compatibility. The AWS-LC team follows the developments around the PQC project and will update the implementation and documentation if necessary. Therefore, AWS-LC can not promise backward compatibility of the Dilithium implementation and API until NIST locks in the specification and reserves the right to change the implementation if necessary.
+
+### **Supported Versions**
+AWS-LC supports only Dilithium3 algorithm at this point. The NID assigned to Dilithium3 is `NID_DILITHIUM3` and the corresponding `PKEY` identifier is `EVP_PKEY_DILITHIUM3`.
+
+### **Source Code Origin and Modifications**
+The source code was taken from the primary source of Diltihium at [link](https://github.com/pq-crystals/dilithium), at [commit](https://github.com/pq-crystals/dilithium/commit/3e9b9f1412f6c7435dbeb4e10692ea58f181ee51) as of 9th May 2022.
+The `api.h`, `fips202.h` and `params.h` header files were modified [in this PR](https://github.com/awslabs/aws-lc/pull/655) to support our [prefixed symbols build](https://github.com/awslabs/aws-lc/blob/main/BUILDING.md#building-with-prefixed-symbols).
+
+Only the reference C implementation of Dilithium3 is currently integrated. The code is in the `pqcrystals-dilithium_dilithium3_ref` folder. The following changes were made to the code.
+
+* `randombytes.{h|c}` are deleted because we are using the randomness generation functions provided by AWS-LC.
+* `rng.{h|c}` are deleted because we are using the randomness generation functions provided by AWS-LC.
+* `symmetric-aes.c` are removed because we are using only the SHA3 based Dilithium.
+* `sign.c`: calls to `randombytes` function is replaced with calls to `pq_custom_randombytes` and the appropriate header file is included (`crypto/rand_extra/pq_custom_randombytes.h`).
+* `sign.c` was modified from `*mlen = -1;` to `*mlen = 0;` within the `crypto_sign_open` function. This was modified as `*mlen = -1;` is attempting to assign `-1` to an unsigned integer, giving the warning: ` warning C4245: '=': conversion from 'int' to 'std::size_t'`. This warning would cause Windows x86 builds to fail. See [pq-crystals/dilithium#65](https://github.com/pq-crystals/dilithium/issues/65) for upstream issue raised.
```

Comment:
not directly relevant to the PR, but @jakemas can you please ping the Dilithium maintainers about the opened issue?

Reply:
Ok